src/Hooks.sol

```diff
@@ -59,12 +89,51 @@ contract Hooks is OwnableUpgradeable, IHooks {
         }
     }
 
+    /**
+     * @notice Callback from the Vault before withdraw
+     * @param assets The amount of assets to be withdrawn
+     * @param user The address of the user withdrawing
+     */
+    function beforeWithdraw(uint256 assets, address user) external onlyVault {
+        Vault vault_ = Vault(payable(address(VAULT)));
+        if (vault_.withdrawalFeeExempted(user)) return;
+        uint256 fees = vault_._feeOnRaw(assets);
+        uint256 sharesToMint = vault_.convertToShares(fees);
+        if (sharesToMint > 0) {
+            VAULT.mintShares(performanceFeeRecipient, sharesToMint);
+            emit WithdrawFeeCharged(performanceFeeRecipient, sharesToMint, fees, assets);
+        }
+    }
+
+    /**
+     * @notice Callback from the Vault after redeem
+     * @param shares The amount of shares to be redeemed
+     * @param user The address of the user redeeming
+     */
+    function afterRedeem(uint256 shares, address user) external onlyVault {
+        Vault vault_ = Vault(payable(address(VAULT)));
+        if (vault_.withdrawalFeeExempted(user)) return;
+        uint256 sharesToMint = vault_._feeOnTotal(shares);
```

Comment:
_feeOnTotal takes as a parameter assets, not obvious why this ok here needs clarity, as shares are passed in